collections/without_all.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/** Used as the size to enable large array optimizations. */
+const LARGE_ARRAY_SIZE = 200;
+
+/**
+ * Returns an array excluding all given values.
+ *
+ * Example:
+ *
+ * ```ts
+ * import { withoutAll } from "./withoutAll.ts";
+ *
+ * const withoutList = withoutAll([2, 1, 2, 3], [1, 2]);
+ *
+ * ```
+ */
+export function withoutAll<T>(
+  array: Array<T>,
+  values: Array<T>,
+): Array<T> {
+  if (array.length >= LARGE_ARRAY_SIZE) {
+    const set = new Set<T>();
+
+    for (const element of array) {
+      set.add(element);
+    }
```

Comment:
I do not think a `LARGE` check is necessary. Building a set **from the `values`** is better runtime in all cases where `values` is longer than 1, at the cost of one `Set` in memory - I would definitely do that.

I think this might be enough to implement the whole thing:

```suggestion
    const toExclude = new Set(values);
    
    return array.filter((it) => toExclude.has(it));
```

Reply:
I'm ok with the above suggested implementation.

If we do care about the memory/speed trade off very much, I guess it could be something like this:
```ts
if (values.length >= LARGE) {
  const toExclude = new Set(values);
  return array.filter((it) => !toExclude.has(it));
}
return array.filter((it) => values.indexOf(it) === -1);
```

Reply:
I mean the memory it takes scales with the input array. So I would argue that it is worth it at any length of `values`, because (with `V` as length of `values` and `N` as length of `array`) it always reduces the runtime from `N*V` to `N + V` (which is very significant) and only takes `V` memory (so not a lot for small `V`s).

Reply:
Sounds reasonable. Ok. Let's simply create a set always